hack/make/.go-autogen

```diff
@@ -18,45 +16,12 @@ LDFLAGS="${LDFLAGS} \
 
 # Compile the Windows resources into the sources
 if [ "$(go env GOOS)" = "windows" ]; then
-	mkdir -p autogen/winresources/tmp autogen/winresources/dockerd
-	cp hack/make/.resources-windows/resources.go autogen/winresources/dockerd/
-
-	if [ "$(go env GOHOSTOS)" == "windows" ]; then
-		WINDRES=windres
-		WINDMC=windmc
-	else
-		# Cross compiling
-		WINDRES=x86_64-w64-mingw32-windres
-		WINDMC=x86_64-w64-mingw32-windmc
-	fi
-
-	# Generate a Windows file version of the form major,minor,patch,build (with any part optional)
-	if [ ! -v VERSION_QUAD ]; then
-		VERSION_QUAD=$(echo -n $VERSION | sed -re 's/^([0-9.]*).*$/\1/' | tr . ,)
-	fi
-
-	# Pass version and commit information into the resource compiler
-	defs=
-	[ ! -z $VERSION ]      && defs="$defs -D DOCKER_VERSION=\"$VERSION\""
-	[ ! -z $VERSION_QUAD ] && defs="$defs -D DOCKER_VERSION_QUAD=$VERSION_QUAD"
-	[ ! -z $GITCOMMIT ]    && defs="$defs -D DOCKER_COMMIT=\"$GITCOMMIT\""
-
-	function makeres {
-		${WINDRES} \
-			-i hack/make/.resources-windows/$1 \
-			-o $3 \
-			-F $2 \
-			--use-temp-file \
-			-I autogen/winresources/tmp \
-			$defs
-	}
-
-	${WINDMC} \
-		hack/make/.resources-windows/event_messages.mc \
-		-h autogen/winresources/tmp \
-		-r autogen/winresources/tmp
-
-	makeres dockerd.rc pe-x86-64 autogen/winresources/dockerd/rsrc_amd64.syso
-
-	rm -r autogen/winresources/tmp
+  if [ ! -x "$(command -v goversioninfo)" ]; then
+    >&2 echo "goversioninfo not found, skipping manifesting binary"
+  else
+    (
+      . hack/make/.mkversioninfo
+      go generate -v "${GO_PACKAGE}"
```

Comment:
> what value is being added by running the `goversioninfo` command indirectly through `go generate`?

Was just to avoid too much scripting and also share the same logic with `go generate` between `.mkversioninfo` and `.go-autogen.ps1`. I we decide to only rely on cross compilation in the future then yes it makes sense to remove the `go generate` logic and have everything in `.mkversioninfo`. See https://github.com/crazy-max/moby/commit/3c5a037b329924251efb2112eb53a66c64d41350